### PR TITLE
feat/DT-2059 Add Table: Upload CSV

### PR DESCRIPTION
**Pull request by lluli**
This is the upload CSV page for the service of adding a table to an existing catalogue page. This work includes validation:

- Given a user hasn’t selected a CSV to upload and they press the continue button then an error message should appear above the text input and an error summary should appear at the top of the page saying “Select a CSV”
- Given a user has selected a CSV that contains special characters in the name and they press the continue button then an error message should appear above the text input and an error summary should appear at the top of the page saying “File name cannot contain special characters apart from underscores and hyphens”

**How to Test**
1. Navigate to /datasets
2. Add /add-table to the end of the dataset ID
3. Click through the flow to get to 'Upload CSV'
4. Try uploading no file then try uploading a .csv file with special characters, both should display the error messages above.

### CODE REVIEW

File: dataworkspace/dataworkspace/apps/datasets/add_table/views.py [hunk 1]
```diff
@@ -155,11 +162,70 @@ def get_context_data(self, **kwargs):
...
+        csv_file.seek(0)
+        try:
+            client.put_object(
+                Body=csv_file,
+                Bucket=settings.NOTEBOOKS_BUCKET,
+                Key=key,
+            )
+        except ClientError as ex:
+            # pylint: disable=raise-missing-from
+            # pylint: disable=no-member
```
Comment (ian-leggett) [inline dataworkspace/dataworkspace/apps/datasets/add_table/views.py:215]: Do we need to override pylint here?

File: dataworkspace/dataworkspace/apps/datasets/manager/views.py [hunk 2]
```diff
@@ -140,6 +140,7 @@ def form_valid(self, form):
                 f"{source.schema}-{source.table}-{datetime.now().isoformat()}",
             )
         except HTTPError:
+            print("HTTPError>>>>>>", HTTPError)
```
Comment (ian-leggett) [inline dataworkspace/dataworkspace/apps/datasets/manager/views.py:143]: Can we remove the print statement.

File: dataworkspace/dataworkspace/apps/your_files/views.py [hunk 3]
```diff
@@ -299,6 +299,7 @@ def get_initial(self):
                     "table_exists_action": self.request.GET.get("table_exists_action"),
                 }
             )
+        print(self.request.GET["path"])
```
Comment (ian-leggett) [inline dataworkspace/dataworkspace/apps/your_files/views.py:302]: Another print statement needs removing.

File: dataworkspace/dataworkspace/templates/datasets/add_table/upload_csv.html [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+<a href="{{ backlink }}" class="govuk-back-link">Back</a>
+{% endblock %}
+{% endif %}
+
+{% block content %}
+<div class="govuk-grid-row">
+    <div class="govuk-grid-column-two-thirds">
+        {% include 'design_system/error_summary.html' with form=form %}
+        <h1 class="govuk-heading-xl">Upload CSV</h1>
+        <h1 class="govuk-heading-l">Before you upload your CSV</h1>
```
Comment (ian-leggett) [inline dataworkspace/dataworkspace/templates/datasets/add_table/upload_csv.html:16]: We have two H1 tags here. I think the second one should be a H2

**Review by ian-leggett — COMMENTED** [review]
Looking good, just a couple of small comments.

File: dataworkspace/dataworkspace/apps/datasets/add_table/views.py (hunk 1 above)
Comment (lluli) [reply dataworkspace/dataworkspace/apps/datasets/add_table/views.py:215]: The first one came over from the code in manage table and the second one was a pylint error saying 'Instance of 'ClientError' has no 'response' member (no-member)'. The flow works correctly so we disabled it. Let me know if this needs to be changed.

**Pull request closed without merging**